Eq.thy

```diff
@@ -101,6 +101,9 @@ end
 lemma inv_type: "\<lbrakk>A: U i; x: A; y: A\<rbrakk> \<Longrightarrow> inv[A, x, y]: x =[A] y \<rightarrow> y =[A] x"
 unfolding inv_def by derive
 
+lemma inv_app_type:"\<lbrakk>A: U i; x: A; y: A;p:x =[A] y\<rbrakk> \<Longrightarrow> inv[A, x, y]`p: y =[A] x"
```

Comment:
This is easily proved from the preceding lemma `inv_type` by `derive`, so I think it unnecessary.